File: gomibako/ruby-book/test/taiyaki_test.rb
'''
Ruby
https://www.youtube.com/watch?v=HsivL1JBRaw
これをPython→Rubyでやってみよう
'''

require 'minitest/autorun'


class Taiyaki
    attr_reader :anko, :size

    def initialize(anko, size)
        @anko = anko
        @size = size
    end
end

class TaiyakiTest < Minitest::Test
    
    def test_taiyaki
        taiyaki_1 = Taiyaki.new('あんこ', 'ふつう')
        assert_equal 'あんこ', taiyaki_1.anko
        assert_equal 'ふつう', taiyaki_1.size
        
        taiyaki_2 = Taiyaki.new('白あん', '大きい')
        assert_equal '白あん', taiyaki_2.anko
        assert_equal '大きい', taiyaki_2.size
    end
end

File: gomibako/ruby-book/test/taiyaki_test.py
'''
Python
https://www.youtube.com/watch?v=HsivL1JBRaw
RubyのこれをPythonに移植してみよう！
'''
import unittest


class Taiyaki:
    '''
    たい焼きの型でござるな
    '''

    def __init__(self, anko):
        self.anko = anko


class TaiyakiTest(unittest.TestCase):
    '''
    単体テストの呼び出しでござるな。
    '''

    def test_taiyaki(self):
        '''
        単体テスト本体でござるな。
        '''
        taiyaki_1 = Taiyaki('あんこ')
        print(taiyaki_1)
        taiyaki_2 = Taiyaki('白あん')
        print(taiyaki_2)


if __name__ == '__main__':
    unittest.main(exit=False)
